api/scpca_portal/test/management/commands/test_create_portal_metadata.py

```diff
@@ -111,3 +111,31 @@ def test_create_portal_metadata(self, mock_upload_output_file):
                     [row[common.LIBRARY_ID_KEY] for row in rows if common.LIBRARY_ID_KEY in row]
                 )
                 self.assertEquals(output_libraries, expected_libraries)
+
+    @patch("scpca_portal.management.commands.create_portal_metadata.s3.upload_output_file")
+    @patch("scpca_portal.management.commands.create_portal_metadata.s3.delete_output_file")
+    def test_purge_computed_file(self, mock_delete_output_file, mock_upload_output_file):
+        # Set up the database for test
+        self.load_test_data()
+        # Create the portal metadata computed_file to purge
+        computed_file = self.processor.create_portal_metadata(
+            clean_up_output_data=False, update_s3=True
+        )
+
+        if computed_file:
+            mock_upload_output_file.assert_called_once_with(computed_file.s3_key)
+            # Query the newly created computed_file with its corresponding id
+            computed_file_before_purge = ComputedFile.objects.filter(
+                id=computed_file.id, portal_metadata_only=True
+            ).first()
+
+            # Purge computed_file
+            self.processor.purge_computed_file(delete_from_s3=True)
+            mock_delete_output_file.assert_called_once_with(computed_file.s3_key)
+            # Make sure that computed_file has been deleted from the database
+            computed_file_after_purge = ComputedFile.objects.filter(
+                id=computed_file_before_purge.id, portal_metadata_only=True
+            ).first()
+            self.assertIsNone(computed_file_after_purge)
+        else:
```

Comment:
`test_only_one_computed_file_at_any_point` could be the test name, and you just need to run the command twice, then ensure that there is only one record in the database for computed files that has `portal_metadata_only=True`

Reply:
See at 9b19c58cad3b77feaafca2d726224d13a8c6d932.